code/core/src/components/components/shared/useLocationHash.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+
+const hashMonitor = {
+  currentHash: globalThis.window?.location.hash ?? '',
+  intervalId: null as ReturnType<typeof setInterval> | null,
+  listeners: new Set<(hash: string) => void>(),
+
+  start() {
+    if (this.intervalId === null) {
+      this.intervalId = setInterval(() => {
```

Comment:
The polling mechanism for retrieving the hash appears to be quite complex. Can't you simply use the hashchange event instead and set up a listener for it?
https://developer.mozilla.org/en-US/docs/Web/API/Window/hashchange_event

Reply:
No, it doesn't fire consistently with our router so I had to resort to this... 😔